app/src/main/java/com/arctouch/codechallenge/repositories/MoviesRepository.kt

```diff
@@ -0,0 +1,125 @@
+package com.arctouch.codechallenge.repositories
+
+import android.arch.lifecycle.MutableLiveData
+import android.util.Log
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.api.TmdbApi
+import com.arctouch.codechallenge.model.Genre
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.helpers.RetrofitHelper
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+
+object MoviesRepository {
+
+    var nextPage: Long = 1
+    var totalPages: Long = 1
+    var searchText: String = ""
+        set(value) {
+            if (field != value) {
+                field = value
+                nextPage = 1
+                totalPages = 1
+                moviesLiveData.value = listOf()
+                fetchMoviesList()
+            }
+        }
+
+    var compositeSubscription = CompositeDisposable()
+
+    private val movies: ArrayList<Movie> = arrayListOf()
+
+    val genresLiveData = MutableLiveData<List<Genre>>()
+    val moviesLiveData = MutableLiveData<List<Movie>>()
+    val selectedMovieLiveData = MutableLiveData<Movie>()
+    val lastPageLiveData = MutableLiveData<Int>()
+    val loadingLiveData = MutableLiveData<Boolean>()
+    val errorLiveData = MutableLiveData<String>()
+
+    fun fetchGenres() {
+        RetrofitHelper.getApi()?.genres(TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE)
+                ?.subscribeOn(Schedulers.io())
+                ?.observeOn(AndroidSchedulers.mainThread())
+                ?.subscribe({
```

Comment:
That does not matter. Since the repository updates the view model which is lifecycle aware and it is not destroyed during the rotation, it would hold the last values set to the LiveData and once the screen reset in the new orientation it would observe the LiveData which would feed its last updated value.